ui/src/components/Bouts/BoutPage.tsx

```diff
@@ -0,0 +1,697 @@
+import {
+  ArrowRight,
+  Clear,
+  GraphicEq,
+  KeyboardDoubleArrowLeft,
+  KeyboardDoubleArrowRight,
+  Launch,
+  Notifications,
+  Start,
+  ZoomIn,
+  ZoomOut,
+} from "@mui/icons-material";
+import {
+  Alert,
+  Button,
+  Chip,
+  Fade,
+  FormControl,
+  FormHelperText,
+  IconButton,
+  InputLabel,
+  ListItemIcon,
+  MenuItem,
+  Select,
+  Tab,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Tabs,
+  useMediaQuery,
+  useTheme,
+} from "@mui/material";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import { addMinutes, format, max, min, subDays, subMinutes } from "date-fns";
+import _ from "lodash";
+import Image from "next/legacy/image";
+import { useRouter } from "next/router";
+import { useCallback, useMemo, useRef, useState } from "react";
+
+import SpectrogramTimeline, {
+  SpectrogramControls,
+} from "@/components/Bouts/SpectrogramTimeline";
+import { BoutPlayer, PlayerControls } from "@/components/Player/BoutPlayer";
+import {
+  AudioCategory,
+  BoutQuery,
+  FeedQuery,
+  useCreateBoutMutation,
+  useDetectionsQuery,
+  useGetCurrentUserQuery,
+  useListFeedStreamsQuery,
+  useUpdateBoutMutation,
+} from "@/graphql/generated";
+import vesselIconImage from "@/public/icons/vessel-purple.svg";
+import wavesIconImage from "@/public/icons/water-waves-blue.svg";
+import whaleFlukeIconImage from "@/public/icons/whale-fluke-gray.svg";
+import { formatTimestamp, roundToNearest } from "@/utils/time";
+
+import LoadingSpinner from "../LoadingSpinner";
+
+export default function BoutPage({
+  isNew,
+  feed,
+  targetAudioCategory,
+  targetTime,
+  bout,
+}: {
+  isNew: boolean;
+  feed: FeedQuery["feed"];
+  targetAudioCategory?: AudioCategory;
+  targetTime?: Date;
+  bout?: BoutQuery["bout"];
+}) {
+  const theme = useTheme();
+  const isDesktop = useMediaQuery(theme.breakpoints.up("sm"));
+  const router = useRouter();
+  const now = useMemo(() => new Date(), []);
+  targetTime =
+    targetTime ?? (bout?.startTime && new Date(bout.startTime)) ?? now;
+
+  const [boutSaved, setBoutSaved] = useState(false);
+
+  const { currentUser } = useGetCurrentUserQuery().data ?? {};
+  const playerTime = useRef<Date>(targetTime);
+  const setPlayerTime = useCallback(
+    (time: Date) => (playerTime.current = time),
+    [],
+  );
+  const [playerControls, setPlayerControls] = useState<PlayerControls>();
+  const spectrogramControls = useRef<SpectrogramControls>();
```

Comment:
PlayerControls is useState, SpectrogramControls is useRef. Seems inconsistent? I remember there were some issues with getting the controls to work nicely, but shouldn't they both have a similar interface?

Reply:
Good point, I can update.